Tab Bar Items "barItems" stored data in localStorage does not persist on page reload

I am currently facing an issue where the data stored for Tab Bar Items "barItems" in localStorage does not persist on page reload. Instead, it gets replaced by the initial state defined as "barItems". I expect the previously stored data to be loaded from localStorage and used as the initial state.

**Code Sample:**
```javascript
const barItems = [{ id: 1, title: "Welcome", active: true }];

const [items, setItems] = useState([]);

useEffect(() => {
    const storedItems = window.localStorage.getItem("barItems");
    if (storedItems) {
      setItems(JSON.parse(storedItems));
    } else {
      setItems(barItems);
    }
}, []);
```

**Expected Behavior:**
1. On the initial load, the code should check if there is any data stored with the key "barItems" in localStorage.
2. If there is stored data, it should be retrieved and set as the initial state for the items state variable.
3. If no data is found in localStorage, the barItems array should be used as the initial state.

**Actual Behavior:**
The data stored in localStorage does not persist on page reload. Instead, the "barItems" array is used as the initial state, overwriting any previously stored data.

**Steps to Reproduce:**

1. Load the page "/".
2. Store some data in localStorage manually with the key "barItems" or add more tab bar with the first editor Navbar Icon .
3. Reload the page.
4. Observe that the stored data is replaced by the barItems array.

**Please let me know if I should work on this issue.**

Good evening @annonymousauthority  I want to take on this issue but I am having issues with running the server, this is the error I am getting. It has something todo with next font.


- error src\app\localFont.js
An error occured in `next/font`.

Error: Cannot find module './lib/stringify'
Require stack:
- C:\Users\user\Documents\P2PCoder\node_modules\braces\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\micromatch\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\utils\pattern.js
- C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\utils\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\managers\tasks.js
- C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\tailwindcss\lib\lib\content.js
- C:\Users\user\Documents\P2PCoder\node_modules\tailwindcss\lib\lib\setupTrackingContext.js
- C:\Users\user\Documents\P2PCoder\node_modules\tailwindcss\lib\plugin.js
- C:\Users\user\Documents\P2PCoder\node_modules\tailwindcss\lib\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\build\webpack\config\blocks\css\plugins.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\build\webpack\config\blocks\css\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\build\webpack\config\index.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\build\webpack-config.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\server\dev\hot-reloader.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\server\dev\next-dev-server.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\server\next.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\server\lib\render-server.js
- C:\Users\user\Documents\P2PCoder\node_modules\next\dist\compiled\jest-worker\processChild.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at C:\Users\user\Documents\P2PCoder\node_modules\next\dist\server\require-hook.js:180:36
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\braces\index.js:3:19)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\micromatch\index.js:4:16)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\utils\pattern.js:6:20)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\utils\index.js:12:17)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\managers\tasks.js:4:15)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\fast-glob\out\index.js:2:21)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\user\Documents\P2PCoder\node_modules\tailwindcss\lib\lib\content.js:23:58)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)

--- a/src/app/page.js
+++ b/src/app/page.js
@@ -9,7 +9,14 @@ const barItems = [{ id: 1, title: "Welcome", active: true }];
 
 function Home() {
   const [items, setItems] = useState([]);
-
+  useEffect(() => {
+    if (items.length == 0) {
+      setItems(barItems);
+    } else {
+      console.log(items);
+      window.localStorage.setItem("barItems", JSON.stringify(items));
+    }
+  }, [items]);
   useEffect(() => {
     const storedItems = window.localStorage.getItem("barItems");
     if (storedItems) {
@@ -18,28 +25,23 @@ function Home() {
       setItems(barItems);
     }
   }, []);
-  useEffect(() => {
-    if (items.length == 0) {
-      setItems(barItems);
-    } else {
-      window.localStorage.setItem("barItems", JSON.stringify(items));
-    }
-  }, [items]);
 
   const handleButtonClicks = (i) => {
     if (i == 0 && items.length < 5) {
-      const newItems = items.map((item) => ({
+      const oldItems = items.map((item) => ({
         ...item,
         active: false,
       }));
-      setItems([
-        ...newItems,
+      const newItems = [
+        ...oldItems,
         {
           id: items.length + 1,
           title: "untitled",
           active: true,
         },
-      ]);
+      ];
+      setItems(newItems);
+      // sortTabs(newItems)
     }
   };
 
@@ -59,6 +61,7 @@ function Home() {
     }));
     newItems.splice(index, 1);
     setItems(newItems);
+    // sortTabs(newItems)
   };
 
   const setActive = (idx, index) => {

--- a/src/app/page.js
+++ b/src/app/page.js
@@ -41,7 +41,6 @@ function Home() {
         },
       ];
       setItems(newItems);
-      // sortTabs(newItems)
     }
   };
 
@@ -61,7 +60,6 @@ function Home() {
     }));
     newItems.splice(index, 1);
     setItems(newItems);
-    // sortTabs(newItems)
   };
 
   const setActive = (idx, index) => {

